### PR TITLE
bpo-31417 Use the enumerate function where appropriate.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
To make code explicit and more readable

Use the enumerate function to replace occurrences of the pattern:

`for i in range(len(sources)):
    src = sources[i]`

https://bugs.python.org/issue31417

https://bugs.python.org/issue31417

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/modulefinder.py [hunk 1]
```diff
@@ -554,9 +554,9 @@ def replace_paths_in_code(self, co):
             self.processed_paths.append(original_filename)
 
         consts = list(co.co_consts)
```
Comment (vstinner) [inline Lib/modulefinder.py:556]: While we modify this function, I suggest to move type(co) out of the loop:

code_type = type(co)

**Review by vstinner — APPROVED** [review]
LGTM, except of a minor comment.

File: Lib/distutils/ccompiler.py [hunk 2]
```diff
@@ -341,8 +341,7 @@ def _setup_compile(self, outdir, macros, incdirs, sources, depends,
         pp_opts = gen_preprocess_options(macros, incdirs)
 
         build = {}
-        for i in range(len(sources)):
-            src = sources[i]
+        for i, src in enumerate(sources):
```
Comment (MSeifert04) [inline Lib/distutils/ccompiler.py:344]: This stell leaves the `obj = objects[i]`, why not just `for src, obj in zip(sources, objects):`

File: Lib/inspect.py [hunk 3]
```diff
@@ -1258,8 +1258,8 @@ def convert(name, locals=locals,
                 formatarg=formatarg, formatvalue=formatvalue):
         return formatarg(name) + formatvalue(locals[name])
     specs = []
-    for i in range(len(args)):
-        specs.append(convert(args[i]))
+    for i, arg in enumerate(args):
```
Comment (MSeifert04) [inline Lib/inspect.py:1261]: Is there a reason to keep the `i` at all here?

Comment (mbyrnepr2) [reply Lib/inspect.py:1261]: This pattern can be seen in places in the repository.
For example, there is a similar pattern in the inspect.py module in the Signature class:
```for idx, param in enumerate(parameters):```
where idx is not subsequently used.

The original code also did not use a standard:
 ```for i in args:```

I focused on replacing range(len(args)) with enumerate(args).

File: Lib/distutils/ccompiler.py (hunk 2 above)
Comment (mbyrnepr2) [reply Lib/distutils/ccompiler.py:344]: I tried to stay close to the original code and do what was needed to replace ```range(len(sources))``` with ```enumerate(sources)```.
Your suggestion could be done but it looks like the comments in the bug tracker would prefer no changes at all.

**Pull request closed without merging**